modules/apps/29-fee/types/metadata.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+// MetadataFromVersion attempts to parse the given string into a fee version Metadata,
+// an error is returned if it fails to do so.
+func MetadataFromVersion(version string) (Metadata, error) {
+	var metadata Metadata
+	err := ModuleCdc.UnmarshalJSON([]byte(version), &metadata)
+	if err != nil {
+		return Metadata{}, ErrInvalidVersion
+	}
+
+	return metadata, nil
+}
```

Comment:
should we upstream this to main?

Reply:
Yea, was going to open an issue to use this in some other places! I can cherrypick the commits to main